src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -781,6 +781,23 @@ protected theorem extractLsb_ofNat (x n : Nat) (hi lo : Nat) :
     (extractLsb' start len x).getLsbD i = (i < len && x.getLsbD (start+i)) := by
   simp [getLsbD, Nat.lt_succ]
 
+@[simp] theorem getMsbD_extractLsb' {start len : Nat} {x : BitVec w} {i : Nat} :
+    (extractLsb' start len x).getMsbD i =
+    (decide (i < len) &&
+    (decide (start + (len - 1 - i) < w) && x.getMsbD (w - 1 - (start + (len - 1 - i))))) := by
+  rcases len with rfl | len
```

Comment:
```suggestion
  rcases len with _ | len
```

No need to use `rfl` here (`rfl` is used to call subst on hypothesis of type `len = 42`, here we are just doing straight cases on a `Nat`)